woocommerce/woocommerce-android
# Prevent Null Appearing in Order Detail Title

## Issue

Fixes #1939. These changes make use of `orEmpty` to prevent `#null` from appearing in the title when the Order Details screen is loading. Instead of `Order #null`, `Order #` will appear until the order number has loaded.

## UI Changes

A side-by-side of "before and after":

<p float="left">

 <img src="https://user-images.githubusercontent.com/2998162/81496999-8bc58380-92b3-11ea-84a0-09f8ef4d550d.gif" width="300" />
 <img src="https://user-images.githubusercontent.com/2998162/81496902-eca08c00-92b2-11ea-9860-ecf04412365a.gif" width="300" />
</p>

## Testing

@shiki came up with the following steps for testing the issue and the fix:

1. Comment out this line in `OrderDetailPresenter`:

https://github.com/woocommerce/woocommerce-android/blob/298724cba992a82cbc22f56df9c66c823a90bfd1/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderDetailPresenter.kt#L132

This emulates the scenario where the order does not exist in the database or the app failed to refresh the orders list in the background.

2. Run/Debug the app.

3. Place the app in the background by pressing the Android home button.

4. Create an order on the web.

5. Wait for the push notification. Sometimes it takes a few seconds.

6. Tap on the push notification.

7. Because of the commented code, the app will fetch the order from the API and you should see the “Order #null” title.

By commenting out the line, you can also just tap on an order in the Orders list to replicate, making this faster for testing.

## Author Checklist:
